scripts/partition_manager.py

```diff
@@ -593,19 +662,59 @@ def replace_app_with_dynamic_partition(d, dynamic_partition_name):
             v = dynamic_partition_name
 
 
+def set_flash_primary_region(pm_config):
+    for v in pm_config.values():
+        if 'region' not in v:
+            v['region'] = 'flash_primary'
+
+
+def fix_syntactic_sugar(pm_config):
+    set_flash_primary_region(pm_config)
+
+
 def main():
+    pm_config = dict()
+
     if len(sys.argv) > 1:
-        static_config = None
+        static_config = dict()
         args = parse_args()
+
+        # Load all requirements in to 'pm_config'
+        load_reqs(pm_config, args.input_files)
+
         if args.static_config:
-            print("Partition Manager using static configuration at " + args.static_config.name)
             static_config = yaml.safe_load(args.static_config)
-        pm_config = get_pm_config(args.input_files, args.flash_start, args.flash_size * 1024, static_config)
-        if args.dynamic_partition:
-            pm_config[args.dynamic_partition.strip()] = pm_config['app']
-            del pm_config['app']
-            replace_app_with_dynamic_partition(pm_config, args.dynamic_partition.strip())
-        write_yaml_out_file(pm_config, args.output)
+            fix_syntactic_sugar(static_config)
+
+            # Delete all statically defined partitions from the pm_config dict.
+            # This is done since all partitions in pm_config will be resolved.
+            for statically_defined_image in static_config.keys():
+                if statically_defined_image in pm_config:
+                    del pm_config[statically_defined_image]
+
+        pm_config['app'] = dict()
+        fix_syntactic_sugar(pm_config)
+
+        # Place region configuration into a dict
+        regions = {x: {k.replace(f'{x}_', ''): v
+                       for k, v in vars(args).items() if k.startswith(x)}
+                   for x in [y.replace('-', '_') for y in args.regions]}  # Replace - with _ to match argparse namespace
+        solution = dict()  # We need a separate dict to allow deletion of keys inside 'get_region_config'
+
+        for region, region_config in regions.items():
+            region_config['name'] = region
+            partitions = {k: v for k, v in pm_config.items() if region in v['region']}
+            static_partitions = {k: v for k, v in static_config.items() if region in v['region']}
+            get_region_config(partitions, region_config, static_partitions)
+            solution.update(partitions)
+
+            if vars(args)[f'{region}_dynamic_partition']:
+                solution[args.dynamic_partition.strip()] = solution['app']
```

Comment:
Does `args.dynamic_partition` still exist?

Reply:
No, updated to `solution[region_config['dynamic_partition'].strip()] = solution['app']`